client/X11/xf_window.c

```diff
@@ -577,7 +577,7 @@ xfWindow* xf_CreateDesktopWindow(xfContext* xfc, char* name, int width,
 		            settings->DesktopPosY);
 	}
 
-	window->floatbar = xf_floatbar_new(xfc, window->handle, window->width);
+	window->floatbar = xf_floatbar_new(xfc, window->handle, xfc->workArea.width);
```

Comment:
This must be the center of the screen that the window is actually on (which might change on multimonitor systems)